connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/application/ReplE2ESuite.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.application
+
+import java.io.{PipedInputStream, PipedOutputStream}
+import java.util.concurrent.{Executors, Semaphore, TimeUnit}
+
+import org.apache.commons.io.output.ByteArrayOutputStream
+
+import org.apache.spark.sql.connect.client.util.RemoteSparkSession
+
+class ReplE2ESuite extends RemoteSparkSession {
```

Comment:
This should be beefed up quite a bit. @rednaxelafx @cloud-fan do we have a comprehensive set of tests somewhere?

Reply:
To clarify, you mean UDF specific tests right?

Reply:
That's odd. What were the test failures? 
Since it an E2E test, both the connect and server jars would need to be packaged before the tests. Could you try running the test after doing a `package`?